appengine/genetics/js/visualization.js

```diff
@@ -581,3 +823,277 @@ Genetics.Visualization.getMouseName = function(mouse, opt_showStats,
   }
   return name;
 };
+
+/**
+ *
+ * @param {Genetics.MouseAvatar} mouseAvatar
+ * @param {string} reason Type of death, either "EXPLODE", "FIGHT",
+ * "OVERPOPULATION", or "RETIRE".
+ * @private
+ */
+Genetics.Visualization.killMouse_ = function(mouseAvatar, reason) {
+  Genetics.Visualization.removeMouseAvatar(mouseAvatar);
+
+  if (reason == 'EXPLOSION') {
+    // The mouse exploded.
+    var explosion = document.createElement('div');
+    explosion.setAttribute('class', 'explode');
+    explosion.style.top = mouseAvatar.element.style.top;
+    explosion.style.left = mouseAvatar.element.style.left;
+    var afterAnimation = function() {
+      explosion.removeEventListener('animationend', afterAnimation, false);
+      Genetics.Visualization.display_.removeChild(explosion);
+    };
+    explosion.addEventListener('animationend', afterAnimation, false);
+    Genetics.Visualization.display_.appendChild(explosion);
+  } else if (reason == 'OVERPOPULATION') {
+    // The mouse died because of overpopulation
+    var kick = document.createElement('div');
+    kick.setAttribute('class', 'kickedOut');
+    kick.style.top = mouseAvatar.element.style.top;
+    kick.style.left = mouseAvatar.element.style.left;
+    var afterAnimation = function() {
+      kick.removeEventListener('animationend', afterAnimation, false);
+      Genetics.Visualization.display_.removeChild(kick);
+    };
+    kick.addEventListener('animationend', afterAnimation, false);
+    Genetics.Visualization.display_.appendChild(kick);
+  } else if (reason == 'RETIRE') {
+    // The mouse died normally.
+    var tombstone = document.createElement('div');
+    tombstone.setAttribute('class', 'retire');
+    tombstone.style.top = mouseAvatar.element.style.top;
+    tombstone.style.left = mouseAvatar.element.style.left;
+    var afterAnimation = function() {
+      tombstone.removeEventListener('animationend', afterAnimation, false);
+      Genetics.Visualization.display_.removeChild(tombstone);
+    };
+    tombstone.addEventListener('animationend', afterAnimation, false);
+    Genetics.Visualization.display_.appendChild(tombstone);
+  }
+
+  // Update statistics to no longer count dead mouse.
+  Genetics.Visualization.mouseSexes_[mouseAvatar.sex] -= 1;
```

Comment:
Why "-= 1" instead of "--"?  A valid reason is if we might change it to "-= 2", but that doesn't appear reasonable for any of these.